config/dbstructure.config.php

```diff
@@ -1059,6 +1059,7 @@
 		"indexes" => [
 			"PRIMARY" => ["id"],
 			"uid_is-default" => ["uid", "is-default"],
+			"pub_keywords" => ["FULLTEXT", "pub_keywords"],
```

Comment:
I'm unsure if all current MySQL implementations do support Fulltext indexes. Could you check that? Otherwise we need some kind of configuration switch.

Reply:
MariaDB and MYSQL support them.

Reply:
Since which version? I know that for a long time there had been no fulltext support for InnoDB. MariaDB had been the first to implement it.

Reply:
"MySQL version 5.6 or later allows you to define a full-text index for a column whose data type is CHAR , VARCHAR or TEXT in MyISAM or InnoDB table type ."

Reply:
On the other hand, the reference to the full-text index is in the code since July 2011, 6 months before it was implemented in MySQL:
https://github.com/friendica/friendica/blame/b39953b858b775ac56cdb15cd45e25348f0a94c3/mod/msearch.php#L17

Reply:
Yeah, at that time we only supported MyISAM:
https://github.com/friendica/friendica/blob/84d12c3e08ae4b6fe228ce881b0c1570175d1667/database.sql#L31

Reply:
Could you add a check like this one?
https://github.com/friendica/friendica/blob/develop/src/Database/DBStructure.php#L200-L202

So that below 5.6.4 this would simply be ignored? Best should be to remove it from the definition array in this situation, I guess.

Reply:
> So that below 5.6.4 this would simply be ignored?

No, it isn't optional, the `/msearch` query will systematically fail if the index isn't present. Since I'm guessing the search speed could benefit from full-text indexing, this is a feature we might want to depend on for the requirements instead.